src/command/expire/expire.c

```diff
@@ -0,0 +1,680 @@
+                                    {
+                                        strLstAdd(localBackupArchiveRententionList, strLstGet(localBackupRetentionList, localIdx));
+                                    }
+                                }
+                            }
+                        }
+                        // Else if there are not enough backups yet globally to start archive expiration then set the archive
+                        // retention to the oldest backup so anything prior to that will be removed as it is not needed but
+                        // everything else is. This is incase there are old archives left around so that they don't stay around
+                        // forever.
+                        else
+                        {
+                            LOG_INFO(
+                                "full backup total < %u - using oldest full backup for %s archive retention",
+                                archiveRetention, strPtr(archiveId));
+                            strLstAdd(localBackupArchiveRententionList, strLstGet(localBackupRetentionList, 0));
+                        }
+
+                        // If no local backups were found as part of retention then set the backup archive retention to the newest
+                        // backup so that the database is fully recoverable (can be recovered from the last backup through pitr)
+                        if (strLstSize(localBackupArchiveRententionList) == 0)
+                        {
+                            strLstAdd(
+                                localBackupArchiveRententionList,
+                                strLstGet(localBackupRetentionList, strLstSize(localBackupRetentionList) - 1));
+                        }
+
+                        // Get the data for the backup selected for retention and all backups associated with this archive id
+                        List *archiveIdBackupList = lstNew(sizeof(InfoBackupData));
+                        InfoBackupData archiveRetentionBackup = {0};
+                        for (unsigned int infoBackupIdx = 0; infoBackupIdx < infoBackupDataTotal(infoBackup); infoBackupIdx++)
+                        {
+                            InfoBackupData archiveIdBackup = infoBackupData(infoBackup, infoBackupIdx);
+
+                            // If this is the backup selected for retention, store its data
+                            if (strCmp(archiveIdBackup.backupLabel, strLstGet(localBackupArchiveRententionList, 0)) == 0)
+                                archiveRetentionBackup = infoBackupData(infoBackup, infoBackupIdx);
+
+                            // If this is a backup associated with this archive Id, then add it to the list to check
+                            if (archiveIdBackup.backupPgId == archivePgId)
+                                lstAdd(archiveIdBackupList, &archiveIdBackup);
+                        }
+
+                        bool removeArchive = false;
```

Comment:
Put this bool between the comment and the if.

Reply:
done